includes/abilities-api/class-wp-abilities-category-registry.php

```diff
@@ -0,0 +1,246 @@
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param array<string,mixed> $args The arguments used to instantiate the category.
+		 * @param string              $slug The slug of the category.
+		 */
+		$args = apply_filters( 'register_ability_category_args', $args, $slug );
+
+		try {
+			// WP_Ability_Category::prepare_properties() will throw an exception if the properties are invalid.
+			$category = new WP_Ability_Category( $slug, $args );
+		} catch ( \InvalidArgumentException $e ) {
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html( $e->getMessage() ),
+				'n.e.x.t'
+			);
+			return null;
+		}
+
+		$this->registered_categories[ $slug ] = $category;
+		return $category;
+	}
+
+	/**
+	 * Unregisters a category.
+	 *
+	 * Do not use this method directly. Instead, use the `wp_unregister_ability_category()` function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @see wp_unregister_ability_category()
+	 *
+	 * @param string $slug The slug of the registered category.
+	 * @return ?\WP_Ability_Category The unregistered category instance on success, null on failure.
+	 */
+	public function unregister( string $slug ): ?WP_Ability_Category {
+		if ( ! $this->is_registered( $slug ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s: Category slug. */
+				sprintf( esc_html__( 'Category "%s" not found.' ), esc_attr( $slug ) ),
+				'n.e.x.t'
+			);
+			return null;
+		}
+
+		$unregistered_category = $this->registered_categories[ $slug ];
+		unset( $this->registered_categories[ $slug ] );
+
+		return $unregistered_category;
+	}
+
+	/**
+	 * Retrieves the list of all registered categories.
+	 *
+	 * Do not use this method directly. Instead, use the `wp_get_ability_categories()` function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @see wp_get_ability_categories()
+	 *
+	 * @return array<string,\WP_Ability_Category> The array of registered categories.
+	 */
+	public function get_all_registered(): array {
+		return $this->registered_categories;
+	}
+
+	/**
+	 * Checks if a category is registered.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $slug The slug of the category.
+	 * @return bool True if the category is registered, false otherwise.
+	 */
+	public function is_registered( string $slug ): bool {
+		return isset( $this->registered_categories[ $slug ] );
+	}
+
+	/**
+	 * Retrieves a registered category.
+	 *
+	 * Do not use this method directly. Instead, use the `wp_get_ability_category()` function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @see wp_get_ability_category()
+	 *
+	 * @param string $slug The slug of the registered category.
+	 * @return ?\WP_Ability_Category The registered category instance, or null if it is not registered.
+	 */
+	public function get_registered( string $slug ): ?WP_Ability_Category {
+		if ( ! $this->is_registered( $slug ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s: Category slug. */
+				sprintf( esc_html__( 'Category "%s" not found.' ), esc_attr( $slug ) ),
```

Comment:
Similar minor note regaring translations:
```suggestion
				/* translators: %s: Ability category slug. */
				sprintf( esc_html__( 'Ability category "%s" not found.' ), esc_attr( $slug ) ),
```